php/vagrant/provision.sh

```diff
@@ -0,0 +1,121 @@
+
+# Install heroku toolbelt
+echo "- Adding heroku toolbelt to packages sources "
+# These shell script snippets are directly taken from heroku installation script
+echo "deb http://toolbelt.heroku.com/ubuntu ./" > /etc/apt/sources.list.d/heroku.list
+# install heroku's release key for package verification
+wget -O- https://toolbelt.heroku.com/apt/release.key 2>&1 | apt-key add -
+
+echo "---------------------------------------------"
+echo "------- Updating package dependencies -------"
+echo "---------------------------------------------"
+apt-get update -y # no need for sudo, and -y is needed to bypass the yes-no
+
+echo "---------------------------------------------"
+echo "-------- Installing packages ----------------"
+echo "---------------------------------------------"
+
+# Additionnal packages :
+#  - man : for documentation
+#  - dos2unix : to avoid CR-LF issues on windows (and that would prevent ~/.bashrc to work properly because \r would be unrecognized)
+#  - heroku-toolbelt : CLI for Heroku
+#  - ruby : needed for heroku toolbelt
+#  - curl : needed to download things
+#  - Apache2 is installed here so we can update its config before the modules install.
+apt-get install -y --no-install-recommends heroku-toolbelt ruby dos2unix man curl apache2
+# Set Apache ServerName.
+echo "ServerName ${SERVER}" >> /etc/apache2/apache2.conf;
+
+# Installing the PHP dev stack
+apt-get install -y --no-install-recommends git postgresql postgresql-contrib phpunit php5 php5-intl php5-pgsql php5-mcrypt php5-sqlite php5-apcu php5-cli
+
+# Install Heroku CLI
+su - vagrant -c "heroku --version > /dev/null 2>&1"
+
+# Install postgresql and setup user and DB
+echo "---------------------------------------------"
+echo "---- Setting up postgresql with ${DB} DB ----"
+echo "---------------------------------------------"
+# Creating superuser (-s) vagrant
+su - postgres -c "createuser -s vagrant"
+su - vagrant -c "createdb ${DB}"
+# Setting default password for vagrant so Cake can connect
+su - vagrant -c "psql -U vagrant -d ${DB} -c \"ALTER USER \\\"vagrant\\\" WITH PASSWORD 'vagrant';\""
+
+# Copying apache config
+echo "---------------------------------------------"
+echo "------- Configuring Apache2 -----------------"
+echo "---------------------------------------------"
+cd /vagrant
+cp vagrant/000-default.conf /etc/apache2/sites-available/000-default.conf
+# Enabling PHP extensions
+php5enmod mcrypt
+# Enabling mod_rewrite
+a2enmod rewrite
+# Restarting apache for changes to take effects
+service apache2 restart
+# Adding user vagrant to www-data group for file permissions
+usermod -a -G www-data vagrant
+
+# Downloading composer in site dir.
+echo "---------------------------------------------"
+echo "- Setting up Composer and Cake dependencies -"
+echo "---------------------------------------------"
+# Installing composer
+if [ ! -e composer.phar ]; then
+  curl -sS https://getcomposer.org/installer | sudo php
```

Comment:
Instead of progress-bar, this starts printing `Downloading x%` text in newlines. We will fix this later.

![screenshot from 2016-06-01 07 20 57 png](https://cloud.githubusercontent.com/assets/5607371/15695932/8adfbd54-27c9-11e6-80c8-4202dc1c36b0.jpg)